### ISSUE TITLE
Cylinder selections can't be used from AtomGroup instances; don't honor PBC

### ISSUE THREAD

**Issue by mnmelo** [MDAnalysis/mdanalysis#274]
I found a couple of bugs in the cylinder selections:
1- They cannot be issued from an AtomGroup instance, only a Universe;
2- PBC are not handled at all, but no warning is given to the user.
................
I also need some help regarding `_unitcell`. Please read my questions at the bottom.

I'm finishing a clean up of this part of the code, which also gets quite a speedup.
(I preselect the atoms in the paralelipiped circumscribing the cylinder, and only then do I perform the more expensive radial search).

PBC treatment now works, and also for triclinic cells. I limited the selection to situations where the cylinder diameter is not larger than any of the x or y dimensions of the box, nor its height larger than the box z (by box I mean the rectangular compact representation -- I'm lacking some more expert terms here, sorry). Otherwise the PBC search would be too painful.

When applying PBC I repack the coordinates in the rectangular compact representation of the box. If the cell is triclinic finding the y and z of this box involves some trigonometry that, for now, is performed every time a cylinder selection is invoked.
I noticed, however, that the GRO Reader generates a `_unitcell` where the lengths of the x, y, z vectors are the first three elements. That saves the redundant calculations,
Unfortunately, it seems that some Readers output `_unitcell` expressed instead as the norms and angles of non-orthogonal vectors. I can't rely on these `_unitcell`.

So, my questions:
- Is there some other property where the cartesian triclinic unit cell information is stored?
- If not, should we have one? Let's say, we keep `_unitcell` for whatever is read from the file, set `_unitcell_cartesian` for the cartesian one, and `_unitcell_norms_angles` for the norms+angles representation?
- Should this be a separate issue?

**Comment by richardjgowers** [comment]
Hey Manuel

You are correct, `_unitcell` is the format specific info, ie how it came in from the trajectory. So this shouldn't be used outside of Timestep (private ish)

`.dimensions` should __always__ give you `[x, y, z, alpha, beta, gamma]` (3 lengths and 3 angles)

Then there's `coordinates.core.triclinic_vectors` which takes `[x, y, z, a, b, g]` and returns the 3x3 representation of the box as vectors.

Adding `Timestep.triclinic_vectors` (or `triclinic_dimensions`?) as a property which returns the 3x3 array would be a nice shortcut (for another issue), ie.
``` python
Timestep.triclinic_dimensions == triclinic_vectors(Timestep.dimensions)
```

I'm interested to review this once it's nearly ready, so put up a pull request when you're ready.

**Comment by mnmelo** [comment]
I need some feedback on this issue:

When running the unit tests I realized my new cylinder selection implementation didn't return the same atoms as the old one. I tracked this to a different interpretation of what the selection precisely does. For example, for the selection `cyzone 5 10 -8 protein`
My interpretation: atoms in the cylinder of radius 5Å and extending 10Å above and 8Å below the CoG of the protein atoms.
The current implementation: atoms in the cylinder of radius 5Å, centered on the CoG of the protein atoms, and extending from the box z of -8Å to the box z 10Å.

So the difference in where the cylinder is z-centered. The docs can be interpreted either way, and the fact that a negative zmin is used as an example there makes it even more intuitive that the selection is relative to the CoG in all respects, not only the xy.

I feel my interpretation to be more intuitive, since all dimensions are relative to the CoG. But I can't really say which of the two is more useful.
Changing the interpretation may, of course, break scripts that use this selection.

Please chip in your opinions on whether to keep the selection as is or change it to the (in my opinion) more intuitive alternative.

**Comment by mnmelo** [comment]
I gave it some more thought and, as it currently is, the selection makes it impossible to specify cylinders that cross the periodic z boundary. Cylinders specified relative to the CoG doesn't suffer from this problem.
I now vote more decidedly for change.

**Comment by richardjgowers** [comment]
Hey

Having the +z and -z in terms of the center of the cylinder makes most sense to me too, so +1
